### ISSUE TITLE
Creating test cases

### ISSUE THREAD

**Issue by bonimy** [Ersanio/SNES-Binary-Utility#7]
## Issue
While there is some explanation on how the program converts tables, there is still some ambiguity. Mostly in odd cases, or how invalid input should be handled.

## Resolution
Provide outputs for the following inputs (assuming default parameters)

* Simple, expected case
```csharp
"db $00,$11";
```

* Incomplete table
```csharp
"db $00";
```

* Empty table
```csharp
"db";
```

* No valid input
```csharp
"Hello world";
```

* Empty string
```csharp
String.Empty;
```

* Null stirng
```csharp
null;
```

* Invalid table format
```csharp
"dw $0012";
```

* Invalid number size
```csharp
"db $FF12";
```

* Invalid number format
```csharp
"db 00,12";
```

* Invalid number separator
```csharp
"db $00.$12";
```

* Comments
```csharp
"db $00,$12; comment";
```

* Table inbetween invalid text
```csharp
"Hello db $00,$12 world";
```

* Table inbetween lines of invalid text
```csharp
"Hello\ndb $00,$12\nworld";
```

* Extra db
```csharp
"db db $00,$12";
```

* Single db on multiple lines
```csharp
"db $00\ndb $12";
```

* Incorrect space padding for table
```csharp
"db$00,$12";
```

* No db prefix
```csharp
"$00,$12";
```

* Two tables on the same line
```csharp
"db $00,$12 db 45,$34";
```

* Two tables on the same line properly separated by a block
```csharp
"db $00,$12 : db 45,$34";
```

* Improper table sizes that can still be a table
```csharp
"db $9900,$6712";
```

* Label before a table
```csharp
"Table: db $00,$12";
```

* Instructions before table properly block separated
```csharp
"RTS : db $00,$12";
```

* Instruction after
```csharp
"db $00,$12 : PHP : PHK : PLB";
```

* Table with a define in it
```csharp
"db $00,!PlayerAddress";
```

* Table with math in it
```csharp
"db $00,$10+2";
```

**Comment by Ersanio** [comment]
## Default parameters:
- Preceding lines: 20
- Output format: db
- Values per line: 8

* Simple, expected case
```csharp
"db $00,$11";
```
Outputs:
```
                    db $00,$11
```

* Incomplete table
```csharp
"db $00";
Outputs:
```
                    db $00
```

* Empty table
```csharp
"db";
```
Error: Invalid data input

* No valid input
```csharp
"Hello world";
```
Error: Invalid data input

* Empty string
```csharp
String.Empty;
```
Error: Invalid data input

* Null stirng
```csharp
null;
```
Error: Invalid data input

* Invalid table format
```csharp
"dw $0012";
```
db, dw, dl and dd all are accepted, so this would parse as:
```
                    db $12,$00
```

* Invalid number size
```csharp
"db $FF12";
```
This is a tricky one. Xkas assembles this as $12. There are two options:
Error: Invalid table value "$FF12"
or:
```
                    db $12
```

* Invalid number format
```csharp
"db 00,12";
```
Error: Invalid data input

* Invalid number separator
```csharp
"db $00.$12";
```
Error: Invalid data input

* Comments
```csharp
"db $00,$12; comment";
```
It would parse as:
```
                    db $00,$12 ; comment
```

* Table inbetween invalid text
```csharp
"Hello db $00,$12 world";
```
Error: Invalid data input

* Table inbetween lines of invalid text
```csharp
"Hello\ndb $00,$12\nworld";
```
Error: Invalid data input

* Extra db
```csharp
"db db $00,$12";
```
Error: Invalid data input

* Single db on multiple lines
```csharp
"db $00\ndb $12";
```
It would parse as:
```
                    db $00,$12 ; comment
```

* Incorrect space padding for table
```csharp
"db$00,$12";
```
Error: Invalid data input

* No db prefix
```csharp
"$00,$12";
```
Error: Invalid data input

* Two tables on the same line
```csharp
"db $00,$12 db 45,$34";
```
Error: Invalid data input

* Two tables on the same line properly separated by a block
```csharp
"db $00,$12 : db 45,$34";
```
Assuming there's a $ before that 45, it would parse as:
```
                    db $00,$12,$45,$34
```
There will be support for block separators, but only for input.

* Improper table sizes that can still be a table
```csharp
"db $9900,$6712";
```
Again, it's a tricky one considering xkas takes the last 2 digits of the value in case of a db.

* Label before a table
```csharp
"Table: db $00,$12";
```
It would parse as:
```
                    db $00,$12
```

* Instructions before table properly block separated
```csharp
"RTS : db $00,$12";
```
Error: Invalid data input

* Instruction after
```csharp
"db $00,$12 : PHP : PHK : PLB";
```
Error: Invalid data input

* Table with a define in it
```csharp
"db $00,!PlayerAddress";
```
Error: Invalid table value "!PlayerAddress"

* Table with math in it
```csharp
"db $00,$10+2";
```
Error: Invalid table value "$10+2"

So basically there are two types of errors:
`Error: Invalid data input`
This is meant as a more broad version of an error in case a table isn't encountered, or if the table structurally is wrong.

`Error: Invalid table value "value"`
This is a more specific error pointing a certain value which was failed to be parsed.

The tool strictly accepts only tables which start with a db, dw, dl, or dd. It can optionally be preceded by a label, and followed by a comment which only can with a `;`

**Comment by bonimy** [comment]
These test cases have been added by c579b4ce8c8eefa2d66d9f52cbd252297d4d69b1
